src/commons/crypto/signers/kmip/internal.rs

```diff
@@ -0,0 +1,776 @@
+use std::{
+    net::TcpStream,
+    ops::Deref,
+    sync::{Arc, RwLock, RwLockReadGuard},
+    time::{Duration, Instant},
+};
+
+use backoff::ExponentialBackoff;
+use bcder::encode::{PrimitiveContent, Values};
+use bytes::Bytes;
+use kmip::{
+    client::{Client, ClientCertificate, ConnectionSettings},
+    types::{
+        common::{KeyMaterial, ObjectType, Operation},
+        response::{ManagedObject, RNGRetrieveResponsePayload},
+    },
+};
+use openssl::ssl::SslStream;
+use r2d2::PooledConnection;
+use rpki::repository::crypto::{
+    signer::KeyError, KeyIdentifier, PublicKey, PublicKeyFormat, Signature, SignatureAlgorithm, Signer,
+};
+
+use crate::commons::{
+    api::Timestamp,
+    crypto::signers::{
+        error::SignerError,
+        kmip::{
+            connpool::ConnectionManager,
+            keymap::{KeyMap, KmipKeyPairIds},
+        },
+        util,
+    },
+};
+
+//------------ Types and constants ------------------------------------------------------------------------------------
```

Comment:
Indeed. One of them, IGNORE_MISSING_CAPABILITIES, is promoted to a server config setting in a later PR, because it needs to be configurable. Given that Krill lets you tweak lots of settings it wouldn't be odd to add more settings to the config file, and it's hard to pick "correct" values for these settings which implies that we or a customer may need to tweak them. But this PR didn't define anything in config because config file support was left for a later PR. However it might be good to create a tracking issue for making these settings configurable.

Reply:
ok, created #726